### PR TITLE
Added getNamespace method to Localization object

### PR DESCRIPTION
**Pull request by janb87**
This allows using of a custom namespace.

In knowledge center we will implement an UnknownLocalizationHandler. This allows us to set the namespace to "ish" instead of tcm.

I'll create an extra pull request which contains the changes regarding TcmId parsing / creation.

### CODE REVIEW

**Comment by janb87** [comment]
In future version of content delivery it will be possible to retrieve the namespace from the 
"http://content:8081/client/v4/content.svc/GetPublicationMappingsFunctionImport(Url='http://localhost:8880/')" api.

File: dxa-framework/dxa-common-api/src/main/java/com/sdl/webapp/common/api/localization/Localization.java [hunk 1]
```diff
@@ -120,5 +120,11 @@
      */
     List<String> getDataFormats();
 
+    /**
+     * Gets the Namespace
+     *
+     * @return
+     */
+    String getNamespace();
```
Comment (janb87) [inline dxa-framework/dxa-common-api/src/main/java/com/sdl/webapp/common/api/localization/Localization.java]: Add default impl to interface

**Comment by rpannekoek** [comment]
The method needs better documentation; I can see from the method name that it "Gets the namespace", but it is not clear at all to a naïve user what that means.  This is actually about setting the _URI scheme_  (see URI spec) for CM URIs, so it would make sense to call it `getUriScheme` or `getCmUriScheme`.

**Comment by janb87** [comment]
@rpannekoek Did the changes you requested. Can you check again?

**Comment by rpannekoek** [comment]
Quite OK now, although I think that the method name is now better/clearer than its documentation: a "TCM URI" is per definition a URI with "tcm" scheme; this method returns the URI scheme used for CM URIs (which can be either "tcm" or "ish").   So, I'm complaining about one letter in your docs. :-)

**Comment by janb87** [comment]
@rpannekoek Updated the docs

**Comment by rpannekoek** [comment]
OK, thanks. Will be merged after DXA 1.7 release, as discussed.

**Comment by azarakovsky** [comment]
Merged into 1.8

**Pull request closed without merging**